apps/src/p5lab/spritelab/coreLibrary.js

```diff
@@ -344,6 +351,17 @@ function whileClickEvent(inputEvent, p5Inst) {
   return callbackArgList;
 }
 
+function whenSpriteCreatedEvent(inputEvent, p5Inst) {
+  let callbackArgList = [];
+  let sprites = newSprites[inputEvent.args.costume] || [];
+  sprites.forEach(sprite => {
+    callbackArgList.push({newSprite: sprite.id});
+  });
+  // Clear newSprites
+  newSprites = {};
```

Comment:
Oh, actually, clearing this here means it would grow unbounded if you don't have a `when sprite created` event. Probably not a huge deal since most spritelab projects don't have that many sprites, but I'll move it just in case.